diff --git a/core/server/apps/subscribers/lib/router.js b/core/server/apps/subscribers/lib/router.js
index 620c95b67f..d9ebb94011 100644
--- a/core/server/apps/subscribers/lib/router.js
+++ b/core/server/apps/subscribers/lib/router.js
@@ -64,6 +64,12 @@ function handleSource(req, res, next) {
     req.body.subscribed_url = santizeUrl(req.body.location);
     req.body.subscribed_referrer = santizeUrl(req.body.referrer);

+    // A form submission will have a referer header. A direct submission via curl typically does not.
+    // The referrer to the page with the subscription form may be empty but must be present in the submitted form data.
+    if (validator.isEmpty(santizeUrl(req.get('referer') || '')) ||
+        req.body.referrer === undefined) {
+        return next(new Error('Oops, something went wrong!'));
+    }
     delete req.body.location;
     delete req.body.referrer;
